modules/EvseManager/Charger.cpp

```diff
@@ -1050,21 +1054,62 @@ void Charger::stopSession() {
     signalEvent(types::evse_manager::SessionEventEnum::SessionFinished);
 }
 
-void Charger::startTransaction() {
+bool Charger::startTransaction() {
     std::lock_guard<std::recursive_mutex> lock(stateMutex);
     stop_transaction_id_tag.clear();
     transaction_active = true;
+
+    // TODO(ddo) client_id, tariff_id, cable_id and user_data are currently not
+    // set.
+    const types::powermeter::TransactionReq req{evse_id, id_token.id_token, "", 0, 0, ""};
+    for (const auto& meter : r_powermeter_billing) {
+        const auto response = meter->call_start_transaction(req);
+        // If we want to start the session but fail, we stop the charging since
+        // we can't bill the customer.
+        if (response.status == types::powermeter::TransactionRequestStatus::UNEXPECTED_ERROR) {
+            EVLOG_error << "Failed to start a transaction on the power meter " << response.error.value_or("");
+            currentState = EvseState::Error;
```

Comment:
Now that the BSP refactoring was merged to main the EvseState::Error does not exist anymore. I will add an error for that in a PR to core.

Reply:
https://github.com/EVerest/everest-core/pull/488

This contains a new error for this case. You can use:

error_handling->raise_powermeter_transaction_start_failed_error("some description");

to raise the error instead of setting currentState. It will be cleared on unplug, I added that already to PR488.